superset/utils/webdriver.py

```diff
@@ -0,0 +1,190 @@
+SELENIUM_CHECK_INTERVAL = 2
+SELENIUM_RETRIES = 5
+SELENIUM_HEADSTART = 3
+
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    from flask_appbuilder.security.sqla.models import User
+
+
+def get_auth_cookies(user: "User") -> List[Dict[Any, Any]]:
+    # Login with the user specified to get the reports
+    with current_app.test_request_context("/login"):
+        login_user(user)
+        # A mock response object to get the cookie information from
+        response = Response()
+        current_app.session_interface.save_session(current_app, session, response)
+
+    cookies = []
+
+    # Set the cookies in the driver
+    for name, value in response.headers:
+        if name.lower() == "set-cookie":
+            cookie = parse_cookie(value)
+            cookies.append(cookie["session"])
+    return cookies
+
+
+def auth_driver(
+    driver: WebDriver,
+    user: "User",
+    auth_cookies_func: Optional[Callable[["User"], List[Dict[Any, Any]]]] = None,
+) -> WebDriver:
+    """
+        Default AuthDriverFuncType type that sets a session cookie flask-login style
+    :return: WebDriver
+    """
+
+    if user:
+        # Set the cookies in the driver
+        for cookie in (auth_cookies_func or get_auth_cookies)(user):
+            info = dict(name="session", value=cookie)
+            driver.add_cookie(info)
+    elif request.cookies:
+        cookies = request.cookies
+        for k, v in cookies.items():
+            cookie = dict(name=k, value=v)
+            driver.add_cookie(cookie)
+    return driver
+
+
+class WebDriverProxy:
+    def __init__(
+        self,
+        driver_type: str,
+        window: Optional[WindowSize] = None,
+        auth_func: Optional[
+            Callable[..., Any]
+        ] = None,  # pylint: disable=bad-whitespace
+        get_cookies_func: Optional[
+            Callable[..., Any]
+        ] = None,  # pylint: disable=bad-whitespace
+    ):
+        self._driver_type = driver_type
+        self._window: WindowSize = window or (800, 600)
+
+        config_auth_func = current_app.config["WEBDRIVER_AUTH_FUNC"] or auth_driver
+        self._auth_func = auth_func or config_auth_func
+
+        config_auth_cookies_func = (
+            current_app.config["WEBDRIVER_AUTH_COOKIES_FUNC"] or get_auth_cookies
+        )
+        self._auth_cookies_func = get_cookies_func or config_auth_cookies_func
+
+    def create(self) -> WebDriver:
+        if self._driver_type == "firefox":
+            driver_class = firefox.webdriver.WebDriver
+            options = firefox.options.Options()
+        elif self._driver_type == "chrome":
+            driver_class = chrome.webdriver.WebDriver
+            options = chrome.options.Options()
+            arg: str = f"--window-size={self._window[0]},{self._window[1]}"
+            options.add_argument(arg)
+            # TODO: 2 lines attempting retina PPI don't seem to be working
+            options.add_argument("--force-device-scale-factor=2.0")
```

Comment:
I think these should be removed their too specific, and let `WEBDRIVER_CONFIGURATION` handle any kind of fine tunning